staging/src/k8s.io/legacy-cloud-providers/azure/cache/azure_cache.go

```diff
@@ -103,6 +103,15 @@ func (t *TimedCache) getInternal(key string) (*AzureCacheEntry, error) {
 	t.Lock.Lock()
 	defer t.Lock.Unlock()
 
+	// Another goroutine might have written the same key.
```

Comment:
good catch... I don't know what this cache is used for, but this seems like a pretty critical correctness issue

cc @feiskyer

Reply:
good catch, and great thanks for fixing the issue.